components/cursor/cursor.hpp

```diff
@@ -21,16 +21,16 @@ namespace components::cursor {
 
     enum class operation_status_t { success = 1, failure = 0 };
 
-    enum class error_code_t {
+    enum class error_code_t : int32_t {
```

Comment:
Do we really need 32 bits here?

Reply:
It is for C# conversion, but i think it will work with less bytes

Reply:
For some weird reason, 32 bits are not necessary for C++, but other types behaves strange in C#...
is_error() function has this code: { return error_.type != error_code_t::none; }, and it works fine with all singed and unsigned types, but C#, which just calls that function returns true even if error_.type is "none" for every type, exept int32_t. And i couldn't figure out why...

Reply:
If it's require by c# fine but plz leave a comment like "We have to use  int32_t because of c# comparability type restrictions"

Reply:
@OrangeSmokingJacket